STYLEGUIDE.adoc

```diff
@@ -18,14 +18,65 @@ The contents of this document are “recommendations”, that is:
 
 === Basic formatting
 
-Use 4 spaces for indentation. Avoid lines over 100 characters. This is codified in `.editorconfig`.
+Use 2 spaces for indentation. Full-indent for `where` is recommended.
+
+Bad:
+
+[source,Haskell]
+----
+f x = y
+ where
+  y = ...
+----
+
+Good:
+
+[source,Haskell]
+----
+f x = y
+  where
+    y = ...
+----
+
+Avoid lines over 100 characters. This is codified in `.editorconfig`.
 
 Format Haskell source with `stylish-haskell` (config in `.stylish-haskell.yaml`).
+This is enforced in the pre-commit hook and in CI.
+Formatting with `stylish-haskell` must be idempotent.
+
+Type signatures should have the double-colon on the same line as the name, separated by one space.
+
+Bad:
+
+[source,Haskell]
+----
+foo
+  :: A -> B
+  -> C
+----
+
+Good:
+
+[source,Haskell]
+----
+foo :: A -> B -> C
+----
+
+Good:
+[source,Haskell]
+----
+foo ::
+  A ->
+  B -> C
+----
 
 ==== Reasoning
 
 The indentation is arbitrary. The line length is longer than the “traditional” 80 characters, but anecdotally Haskell code tends toward longer lines, and breaking them up doesn’t help readability much.
 
+Placing the double-colon on the same line makes it easier to find definitions.
+We don't necessarily have HLS running all the time, and it may not work on every module.
```

Comment:
At least for me that takes a few more seconds and key strokes though :)

Reply:
True. I personally virtually never do `git grep <definition>`, because I can just use jump-to-definition and I don't need HLS for that. Works even in a GHC Core file for me (it wasn't worth the pain of setting it up though).

Reply:
Yes, not to mention when reading PR in a browser etc.